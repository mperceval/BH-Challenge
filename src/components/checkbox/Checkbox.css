input[type=checkbox] {
    visibility: hidden;
}

.checkbox {
    width: 40px;
    height: 40px;
    background: #ddd;
    margin: 20px 90px;

    border-radius: 100%;
    position: relative;
    box-shadow: 0px 1px 3px rgba(0,0,0,0.5);
}

.checkbox label {
    display: block;
    width: 30px;
    height: 30px;
    border-radius: 100px;
    transition: all .5s ease;
    cursor: pointer;
    position: absolute;
    top: 5px;
    left: 5px;
    z-index: 1;

    background: #333;
    box-shadow:inset 0px 1px 3px rgba(0,0,0,0.5);
}

.checkbox input[type=checkbox]:checked + label {
    background: #26ca28;
}

.checkbox-text-label {
    position: absolute;
    top: 13px;
    left: 55px;
    width: 100px;
}