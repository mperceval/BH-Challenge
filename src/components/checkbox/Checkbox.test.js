import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './index';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Checkbox name='test-checkbox' label='Test Label' value='test-value' initialChecked={true} />);
});

it('renders a checkbox', () => {
  expect(wrapper).toMatchSnapshot();
});

it('sets the initial props on the component', () => {
  const { props } = wrapper.instance();
  expect(props.label).toEqual('Test Label');
  expect(props.value).toEqual('test-value');
  expect(props.name).toEqual('test-checkbox');
});

it('sets the initial state on the component', () => {
  const { state } = wrapper.instance();
  expect(state.checked).toEqual(true);
});

it('updates state if component is clicked', () => {
  const checkbox = wrapper.find('[type="checkbox"]');
  checkbox.simulate('change');
  const { state } = wrapper.instance();
  expect(state.checked).toEqual(false);
});
