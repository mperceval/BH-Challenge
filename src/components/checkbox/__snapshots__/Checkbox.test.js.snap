// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a checkbox 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    initialChecked={true}
    label="Test Label"
    name="test-checkbox"
    value="test-value"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={true}
          id="test-checkbox"
          name="test-checkbox"
          onChange={[Function]}
          type="checkbox"
          value="test-value"
/>,
        <label
          htmlFor="test-checkbox"
/>,
        <span
          className="checkbox-text-label"
>
          Test Label
</span>,
      ],
      "className": "checkbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "id": "test-checkbox",
          "name": "test-checkbox",
          "onChange": [Function],
          "type": "checkbox",
          "value": "test-value",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "htmlFor": "test-checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Label",
          "className": "checkbox-text-label",
        },
        "ref": null,
        "rendered": "Test Label",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={true}
            id="test-checkbox"
            name="test-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="test-value"
/>,
          <label
            htmlFor="test-checkbox"
/>,
          <span
            className="checkbox-text-label"
>
            Test Label
</span>,
        ],
        "className": "checkbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "id": "test-checkbox",
            "name": "test-checkbox",
            "onChange": [Function],
            "type": "checkbox",
            "value": "test-value",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "htmlFor": "test-checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Label",
            "className": "checkbox-text-label",
          },
          "ref": null,
          "rendered": "Test Label",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
